### PR TITLE
Document bootstrap config options and tooling

### PR DESCRIPTION
**Pull request by aloubyansky**
Fix for #2353

### CODE REVIEW

File: docs/src/main/asciidoc/maven-tooling.adoc [hunk 1]
```diff
@@ -213,6 +213,38 @@ In a separated terminal or the embedded terminal, go to the project root and run
...
+            <plugin>
+                <groupId>io.quarkus</groupId>
+                <artifactId>quarkus-bootstrap-maven-plugin</artifactId>
+                <version>${quarkus.version}</version>
+            </plugin>
+----
+
+Now you should be able to execute `mvn quarkus-bootstrap:build-tree` on your project and see something like:
+
+[source,shell]
```
Comment (gsmet) [inline docs/src/main/asciidoc/maven-tooling.adoc]: Maybe you can use `,subs=attributes+` here and use `{quarkus-version}` below?

**Review by gsmet — APPROVED** [review]
@aloubyansky looks mostly good. I added a few minor comments. I let you take a look and fix them before merging if you think it's appropriate?

File: docs/src/main/asciidoc/maven-tooling.adoc [hunk 2]
```diff
@@ -213,6 +213,38 @@ In a separated terminal or the embedded terminal, go to the project root and run
 
 Open the project directory in VS Code. If you have installed the Java Extension Pack (grouping a set of Java extensions), the project is loaded as a Maven project.
 
+== Logging Quarkus application build classpath tree
+
+Usually, dependencies of an application (which is a Maven project) could be displayed using `mvn dependency:tree` command. In case of a Quarkus application, however, this command will list only the runtime dependencies of the application.
+Given that the Quarkus build process adds deployment dependencies of the extensions used in the application to the original application classpath, it could be useful to know which dependencies and which versions end up on the build classpath.
+Luckily, `quarkus-bootstrap` maven plugin includes `build-tree` goal which displays the build dependency tree for the application.
```
[the remark this answers is not in the document]
Comment (aloubyansky) [reply docs/src/main/asciidoc/maven-tooling.adoc]: Why "**a** `build-tree`"? build-tree is a name of the goal. As well as quarkus-bootstrap.

[the remark this answers is not in the document]
Comment (gsmet) [reply docs/src/main/asciidoc/maven-tooling.adoc]: I think it reads better with an article. "It includes a goal."

[the remark this answers is not in the document]
Comment (aloubyansky) [reply docs/src/main/asciidoc/maven-tooling.adoc]: But it's not "It includes a goal." It names the goal.

[the remark this answers is not in the document]
Comment (gsmet) [reply docs/src/main/asciidoc/maven-tooling.adoc]: Either `a` or `the` would read better IMHO. But feel free to merge as is. I approved it :)

**Pull request merged**